costmodel/costmodel.go

```diff
@@ -1110,10 +1104,12 @@ func getNodeCost(cache ClusterCache, cp costAnalyzerCloud.Provider) (map[string]
 		nodes[name] = &newCnode
 	}
 
+	cp.ApplyReservedInstancePricing(nodes)
```

Comment:
There were a few options here regarding how this should work. `ApplyReservedInstancePricing` requires VCPU and RAM values to be set for the `Node` for GCP (AWS will use machine type, but can still be applied here). See my notes above about the parent method `getNodeCost`. It may be better to change `ApplyReservedInstancePricing` to something more generic like `ApplyNodePricing` where each `Provider` could apply (or not) any special pricing schemes (including reserved instances).